src/effects/VST/VSTEffect.h

```diff
@@ -212,15 +350,24 @@ using VSTEffectArray = std::vector < std::unique_ptr<VSTEffect> > ;
 DECLARE_LOCAL_EVENT_TYPE(EVT_SIZEWINDOW, -1);
 DECLARE_LOCAL_EVENT_TYPE(EVT_UPDATEDISPLAY, -1);
 
+
+// Temporary class allowing an intermediate phase in the transformation.
+// It is needed because the instance constructor needs something which is
+// both a PerTrackEffect and a StatefulEffectBase
+class TempClass : public PerTrackEffect
```

Comment:
The existing `StatefulEffect` class might also serve the purpose

Reply:
Hmm... StatefulEffect does not inherit PerTrackEffect. I gave it a go anyway, and I got compiler errors. Never mind, this is only a temporary class that should disappear at point 5.